### ISSUE TITLE
create release and upload files to release

### ISSUE THREAD

**Issue by tbr11** [vmware-tanzu/cartographer#7]
As an operator
I want an easy deployment of kontinue

- [x] when a git tag is seen, build a release and publish it on github (using actions)
- [x] action publishes the images
- [x] the release has a release.yml, which no longer includes cert-manager
- [x] remove release.yml from the repo
- [x] the action publishes shasums
- [x] make sure the readme and contributing still make sense.
- [ ] remove make target release and release bundle from makefile

See this release for an example. https://github.com/vmware-tanzu/carvel-kapp-controller/releases/tag/v0.24.0

**Comment by tbr11** [comment]
We've started work on this using [act](https://github.com/nektos/act/issues/179) to run github actions locally for iteration. Currently `act` does not support triggering on pushed tags. There's a [pull request](https://github.com/nektos/act/pull/735) available to support that feature that works well enough for our iteration, and will hopefully be merged soon.

The fork lets us use an event file that looks like 
```json
{
    "ref": "refs/tags/v0.0.4"
}
```
in order to trigger a tag based workflow with a command like `act -e ~/tag.json --secret-file ~/secret-file`.

We discovered that the official github action to upload releases is deprecated. We could find a third party action, or we can look further into goreleaser.

Goreleaser by default will upload tarballs of go binaries combined with README and LICENSE. We want to instead upload releases to dockerhub publish the release.yaml to github. We can take a look at [running a script in the goreleaser `post` hook](https://goreleaser.com/customization/build/#build-hooks) to do the docker image upload and generate release.yaml. That way we can keep using goreleaser's github release creation including the templating of the changelog.
